### PR TITLE
nick/payment-success-screen

### PR DESCRIPTION
**Pull request by niwong**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?

[//]: # '####### YOUR ANSWER BELOW ###########'

Implemented mid-fi payment success screen for general owners buying shares and subscribers paying bills. 

[//]: # '############## END ##################'

### How to Review

Make a payment! Either pay a bill or buy a share.

Checkout `billPayment.js`, `BuyShares.js` and the new component I made `PaymentSuccessModal` in the shared components modal. I also changed a redux refreshState method in lib with the help of Aivant.

[//]: # 'The order in which to review files and 
what to expect when testing locally'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Related PRs

[//]: # "Optional - related PRs you're waiting on
/ PRs that will conflict, etc"
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Next steps

[//]: # "What doesn't work yet, what's NOT in this 
PR/has to be done "
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Screenshots

[//]: # '#### YOUR SCREENSHOTS BELOW ########'
[//]: # '############## END ##################'

### Design Status

[//]: # 'If this is a frontend PR, what is the expected 
status of the UI in this PR (lo, mid, high- fi?'
[//]: # '####### LOFI/MIDFI/HIFI ###########'
[//]: # '############## END ##################'

CC: @dfangshuo

### CODE REVIEW

File: src/lib/userDataUtils.js [hunk 1]
```diff
@@ -18,10 +18,12 @@ import {
 import { getCredentials } from './credentials';
 
 // Function takes in an ownerId and fetches the latest owner object and all associated user data
-const refreshUserData = async ownerId => {
-  // Save loading status to Redux
-  store.dispatch(setLoadingForUserData());
-  store.dispatch(setLoadingForAnnouncements());
+const refreshUserData = async (ownerId, loadSilently = false) => {
```
Comment (dfangshuo) [inline src/lib/userDataUtils.js:21]: is tthere ar eason loadSilently is set to false here?

File: src/screens/subscriber/components/BillingPayment.js [hunk 2]
```diff
@@ -49,15 +51,24 @@ class BillingPayment extends React.Component {
 
   onPaymentSuccess = async (details, data) => {
     const { activeBill } = this.state;
-    const { history, owner } = this.props;
-    this.setState({ loadingPayment: false, loading: true });
+    const { owner } = this.props;
+    this.setState({
+      loadingPayment: false,
+      transactionAmount: details.purchase_units[0].amount.value,
```
Comment (dfangshuo) [inline src/screens/subscriber/components/BillingPayment.js:57]: same comment as above

File: src/screens/general/BuyShares.js [hunk 3]
```diff
@@ -18,22 +19,29 @@ class BuyShares extends React.PureComponent {
...
   }
 
   onPaymentSuccess = async (details, data) => {
-    const { owner, history } = this.props;
+    const { owner } = this.props;
     const { sharesBuying } = this.state;
-    this.setState({ loading: true });
+
+    this.setState({
+      transactionAmount: details.purchase_units[0].amount.value,
```
Comment (dfangshuo) [inline src/screens/general/BuyShares.js:33]: there's a potential to crash the app here if unexpected data is returned. I make an if check to check that

1. purchase_units is not null
2. purchase_units has length >= 1
if these conditions aren't met, add a console.warn statement

File: src/lib/userDataUtils.js (hunk 1 above)
Comment (niwong) [reply src/lib/userDataUtils.js:21]: Yes, in order to present the user current data after they navigated out of the payment success screen, we still needed to call `refreshUserData` but without the page reloading functionality, so we created this loadSilently option. It still refreshes data, but it doesn't force the page to reload.

File: src/screens/general/BuyShares.js (hunk 3 above)
Comment (niwong) [reply src/screens/general/BuyShares.js:33]: I added the check, but can we have an OnPaymentSuccess with a purchase of zero shares?

File: src/lib/userDataUtils.js (hunk 1 above)
Comment (dfangshuo) [reply src/lib/userDataUtils.js:21]: sounds good! @aivantg ++ for great APIs

File: src/screens/general/BuyShares.js (hunk 3 above)
Comment (dfangshuo) [reply src/screens/general/BuyShares.js]: that is a great point, and i think points down to some design/architectural issues.

In general, I do not think it's a good idea to have to implicitly depend on the fact that the caller of the function will pass this function well-formed input. If we want to go with that pattern, we should enforce this pattern explicitly assertion statements + try/catches.

File: src/screens/shared/components/PaymentSuccessModal.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        >
+          <circle
+            cx="41.5"
+            cy="41.5"
+            r="39"
+            stroke="#CD6795"
+            strokeWidth="5"
+          />
+          <path
+            d="M26.7522 41.7906C25.7674 40.8229 24.1845 40.8367 23.2168 41.8216C22.2491 42.8064 22.263 44.3893 23.2478 45.357L26.7522 41.7906ZM34.593 53L32.8408 54.7832L34.593 56.5049L36.3452 54.7832L34.593 53ZM59.7522 31.7832C60.737 30.8155 60.7509 29.2326 59.7832 28.2478C58.8155 27.263 57.2326 27.2491 56.2478 28.2168L59.7522 31.7832ZM23.2478 45.357L32.8408 54.7832L36.3452 51.2168L26.7522 41.7906L23.2478 45.357ZM36.3452 54.7832L59.7522 31.7832L56.2478 28.2168L32.8408 51.2168L36.3452 54.7832Z"
```
Comment (dfangshuo) [inline src/screens/shared/components/PaymentSuccessModal.js:28]: what is this :o

File: src/screens/shared/components/PaymentSuccessModal.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            fill="#CD6795"
```
Comment (dfangshuo) [inline src/screens/shared/components/PaymentSuccessModal.js:29]: declare this as a constant at the top of the file

**Pull request merged**